syntaxes/styled-components.json

```diff
@@ -102,41 +102,152 @@
-      "begin":
-        "([mM][eE][dD][iI][aA])\\.([[:alpha:]][[:alnum:]]*)(?:(?:\\(([`'\"].*[`'\"])\\))|(?:\\(([[:alnum:]]*)\\))|(?:\\(\\))|(?:\\((.*)\\))){0,1}\\s{0,1}(`)",
-      "beginCaptures": {
-        "1": {
-          "name": "entity.name.function.tagged-template.js"
+            "name": "media-functions",
+            "begin":
+            "([mM][eE][dD][iI][aA])\\.([[:alpha:]][[:alnum:]]*)(?:(?:\\(([`'\"].*[`'\"])\\))|(?:\\(([[:alnum:]]*)\\))|(?:\\(\\))|(?:\\((.*)\\))){0,1}\\s{0,1}(`)",
+            "beginCaptures": {
+                "1": {
+                    "name": "entity.name.function.tagged-template.js"
+                },
+                "2": {
+                    "name": "js"
+                },
+                "3": {
+                    "name": "string.js"
+                },
+                "4": {
+                    "name": "constant.numeric.js"
+                },
+                "5": {
+                    "name": "variable.other.object.js"
+                },
+                "6": {
+                    "name": "punctuation.definition.string.template.begin.js"
+                }
+            },
+            "end": "`",
+            "endCaptures": {
+                "0": {
+                    "name": "punctuation.definition.string.template.end.js"
+                }
+            },
+            "patterns": [
+                {
+                    "include": "source.css.styled"
+                }
+            ]
         },
-        "2": {
-          "name": "js"
-        },
-        "3": {
-          "name": "string.js"
-        },
-        "4": {
-          "name": "constant.numeric.js"
-        },
-        "5": {
-          "name": "variable.other.object.js"
-        },
-        "6": {
-          "name": "punctuation.definition.string.template.begin.js"
-        }
-      },
-      "end": "`",
-      "endCaptures": {
-        "0": {
-          "name": "punctuation.definition.string.template.end.js"
-        }
-      },
-      "patterns": [
         {
-          "include": "source.css.styled"
-        }
-      ]
+            "name": "styled.attrs.object",
+            "begin": "([sS][tT][yY][lL][eE][dD])\\.([[:alpha:]]+)\\.(attrs)\\(",
+            "beginCaptures": {
+                "1": {
+                    "name": "entity.name.function.tagged-template.js"
+                },
+                "2": {
+                    "name": "entity.name.tag.attrs-tag.js"
+                },
+                "3": {
+                    "name": "entity.name.function.tagged-template.js"
+                }
+            },
+            "end": "(?<!\\)|\\)\\s)(`)",
+            "endCaptures": {
+                "1": {
+                    "name": "punctuation.definition.string.template.end.js"
+                }
+            },
+            "patterns": [
+                {
+                    "name": "styled.attrs.objc",
+                    "begin": "(\\{)",
+                        "end": "(\\})",
```

Comment:
The indentation seems to be a bit off here.

Reply:
Yeah I used my auto indenter to make my indent be 4 to match yours (I prefer 2) and I think that it got out of whack. I'm make a quick commit fixing the indentation.